solidity/dashboard/src/services/token-staking.service.js

```diff
@@ -118,31 +121,31 @@ const fetchPendingUndelegation = async (web3Context) => {
     ),
   ])
 
-  const { undelegatedAt } = delegation
+  const { undelegatedAt, createdAt, amount } = delegation
 
   const isUndelegation = delegation.undelegatedAt !== "0"
   const pendingUnstakeBalance = isUndelegation ? delegation.amount : 0
   const undelegationCompletedAt = isUndelegation
     ? moment.unix(undelegatedAt).add(undelegationPeriod, "seconds")
     : null
-  let undelegationStatus
-  if (isUndelegation) {
-    undelegationStatus = undelegationCompletedAt.isBefore(moment())
-      ? COMPLETE_STATUS
-      : PENDING_STATUS
-  } else if (
-    delegation.undelegatedAt === "0" &&
-    delegation.createdAt !== "0" &&
-    delegation.amount === "0"
-  ) {
-    undelegationStatus = COMPLETE_STATUS
+
+  let delegationStatus
```

Comment:
Don't we need another status here "Pending"? What I noticed is when we undelegate Tokens, status is still "Processing" in the Operations, but in the Tokens it's complete for the same undelegation.
![Screenshot 2020-05-25 at 15 14 35](https://user-images.githubusercontent.com/3156420/82815930-7eef8500-9e9a-11ea-809c-6718fbff490e.png)
![Screenshot 2020-05-25 at 15 13 35](https://user-images.githubusercontent.com/3156420/82815943-84e56600-9e9a-11ea-9b9c-eaf5cd919b28.png)

Reply:
From the operator's perspective, it is still `processing` because it has not yet been recovered. But I am not sure if this is the correct approach.

Reply:
@Shadowfiend just to double check. Is this what we want (keeping processing status in Operations when it wasn't recovered yet)?